docs/dg/learningBasics.md

```diff
@@ -0,0 +1,123 @@
+1. You may also find the [Java API](https://docs.oracle.com/javase/8/docs/api/) quite helpful when you are unsure about the usage of a certain class or method in a Java program.
+1. Once you are familiar with the general syntax, you may wish to learn more [advanced topics](https://se-education.org/learningresources/index.html) such as concurrency, synchronization, streams, and writing backend tests in order to work on issues that require the corresponding knowledge.  
+
+{{ step(2) }} Learn the RepoSense backend architecture
+
+You may want to refer to the [backend architecture](architecture.html). This can greatly help you to understand the implementation logic.
+
+To gain a more concrete idea of how the backend works, you can use the IDE Debugger to trace through the steps of how the JSON files are generated by taking the following steps.
+
+<box type="warning" seamless>
+
+The steps are for **Intellij**. If you are using a different IDE, you may need to check the documentation of how to use the debugger separately.
+</box>
+
+ ![tracing-using-debugger](../images/trace-using-debugger.jpg)
+
+1. Check the [debugging guide](https://www.jetbrains.com/help/idea/debugging-your-first-java-application.html) if you are not familiar with debugging in **Intelij**. In particular, learn how to use features such as `step into` and `step over` to trace the program execution.
+1. In `RepoSense.java`, set appropriate break points. For example, you can set it according to the diagram above. The first 3 breakpoints are set to inspect how CLI arguments, CSV files configuration, and JSON file configuration are retrieved, corresponding to the first step in the architecture diagram. The last break point is set at the `generateReposReport` method, where the bulk of work is conducted, including cloning the repository, analyzing the repository, and generating the JSON files to be used by the report.
+1. In `Edit Configurations` of **Intellij**, set the current task to be the Gradle task `RepoSense [run]`, and click the debugging button on the right to start debugging (Make sure that you import the project as a Gradle project first according to [the tutorial](https://se-education.org/guides/tutorials/intellijImportGradleProject.html)).
+1. Step into each break point when you think it is necessary to know more about the implementation, and step over otherwise to continue the execution.
+
+<box type="info" seamless>
+
+When tracing through the program execution, you can cross reference the architecture diagram and Javadoc of the class and method to check your understanding of the procedure.
+</box>
+
+{{ step(3) }} Get started with contributing
+
+Here are some small [tasks](https://github.com/reposense/RepoSense/issues?q=is%3Aissue+label%3Ad.FirstTimers+label%3Aa-Backend+is%3Aopen) for you to get started on contributing to RepoSense backend. Select an issue and try to work on it according to the guidance given in the issue comment.
```

Comment:
Acutally, will it be suitable to pin a specific first timer issue and advice the developer to just try it locally without submitting the pull request, or closing the pull request after submitting it in order for the developer to just gain some experience?

Reply:
These tasks need not be PRs. They can be as simple as modifying some UI element. They can be designed to expose the student to various important parts of the code base. The person should do them locally to self-test knowledge. If they encounter problems and need help, can push to the fork and ask for help by posting in our discussion board. We can give hints or even the model solution inside a panel so that the person can self-solve the problem without even bothering us.
BTW, we can also call them 'tutorials'.

Reply:
Thanks for the suggestion. I will search for tasks along with hint and suggested solution and add them to the section.

Reply:
You can get help from @gerhean  and @dcshzj as well, because different members are more knowledgeable about different parts of the codebase.

Reply:
Sure, no problem